### PR TITLE
Issue #110 - Updated usage prompt.

### LINKED ISSUE #110: No <cmd> advice on cfn-lint --help
The help section should show some examples of how to use the command. Including what options for <cmd> there are.

```bash
user@a55ae7f0fda6:/# cfn-lint --help

  Usage: index [options] <cmd> <file>


  Options:

    -V, --version                        output the version number
    -p, --parameters <items>             List of params
    -p, --pseudo <items>                 List of pseudo overrides
    -G, --no-guess-parameters            Fail validation if a parameter with no Default is not passed
    -g, --only-guess-parameters <items>  Guess the provided parameters, and fail validation if a parameter with no Default is passed
    -h, --help                           output usage information
```

### PR DESCRIPTION
**Pull request by RazzM13**
Fixes #110 by updating "--help" to account for the supported commands and to be shown by default in cases where the required arguments are not provided.

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -5,68 +5,73 @@
...
-    console.error('no command given!');
+function doNoArgument() {
+    console.error('\nmissing required argument');
     process.exit(1);
 }
 
-import validatorBaseImport = require('./validator');
-import docsBaseImport = require('./docs');
+program
+.version(version)
```
Comment (martysweet) [inline src/index.ts]: Can all these .something be indented so it's clear?

File: src/index.ts [hunk 2]
```diff
@@ -5,68 +5,73 @@
...
-    .option('-v, --verbose', 'Verbose error messages')
-    .action(function (arg1, arg2) {
-        firstArg = arg1;
-        secondArg = arg2;
-    });
-
-
-program.parse(process.argv);
+function doNoCommand() {
+    console.error('\nno command given!');
```
Comment (martysweet) [inline src/index.ts]: Lets format this nicely

File: src/index.ts [hunk 3]
```diff
@@ -5,68 +5,73 @@
...
+    process.exit(1);
+}
 
-if (typeof firstArg === 'undefined') {
-    console.error('no command given!');
+function doNoArgument() {
+    console.error('\nmissing required argument');
```
Comment (martysweet) [inline src/index.ts]: Lets format this nicely

File: src/index.ts (hunk 2 above)
Comment (RazzM13) [reply src/index.ts]: Sure, but I'm not really sure what you're referring to? :)

File: src/index.ts (hunk 3 above)
Comment (RazzM13) [reply src/index.ts]: Ditto.

File: src/index.ts (hunk 2 above)
Comment (martysweet) [reply src/index.ts]: Sorry, can we have better English on these feedback messages? As you mentioned in the issue

**Comment by RazzM13** [comment]
Hi @martysweet, I've just noticed two small issues that also occur in `master`:
1) The `-V` option reports a version of `0.0.0` because that is what's specified in `package.json`;
2) The `-p` option can't be used to provide *--pseudo* parameters, as depicted in the usage text, due to the fact that it's currently being assigned to *--parameters*.

Should we treat these defects here, and if so then how? or should I open separate issues?

**Comment by RazzM13** [comment]
Hi @martysweet, I've just noticed a few small considerations that also occur in `master`:
1) The `-V` option reports a version of `0.0.0` because that is what's specified in `package.json`;
2) The `-p` option can't be used to provide *--pseudo* parameters, as depicted in the usage text, due to the fact that it's currently being assigned to *--parameters*.
3) The `--guess-parameters` option doesn't provide a short form.

Should we treat these defects here, and if so then how? or should I open separate issues?

**Comment by martysweet** [comment]
1. I don't want to go through the pain of keeping a version number in master, during build it is set to the correct version (depending on the v1.2.3 version git tag). We could attempt to get from git history but it wouldn't strictly be that version (as it might have commits which haven't been pushed to npm yet) - so it's probably best to leave imho.

2. Woops. We should probably fix that in a backward compatible manner, updating the usage docs as appropriate. 

3. We could add this.

Regarding new issues, has number 2 changed operation since this refactor or has it also been wrong in the docs?

**Comment by RazzM13** [comment]
The second issue was already occuring, as are all the rest. You can clearly see it from:
https://github.com/martysweet/cfn-lint/blob/master/src/index.ts#L34-L35

**Comment by martysweet** [comment]
Ok, I'm going to merge this into master. 

Let's make an issue and fix the clashing `-p` and in that issue lets also add a shorthand for `--guess-parameters`

**Pull request merged**